### ISSUE TITLE
FR: Add twig variables to output

### ISSUE THREAD

**Issue by khalwat** [madebykind/craft.labelenvironment#4]
I'd like to be able to output the name of the current database in the banner.  From a broader POV, if the string passed in was parsed as a template, it'd let people do some very interesting/useful things.

**Comment by michaelrog** [comment]
Good idea. So, we can add a `parseAsTemplate` config to turn it on (extra processing not needed by default), and then conditionally run text through `parseObjectTemplate` in:
- `EnvironmentLabelService->getLabel()`
- `EnvironmentLabelService->getPrefix()`
- `EnvironmentLabelService->getSuffix()`

One potential security caveat: The label _is_ shown on the admin login screen. (I suppose it's up to each dev to protect their CP/info as needed.)

**Comment by tomdavies** [comment]
@khalwat I agree this is a worthwhile feature, and have pushed an initial implementation to the `develop` branch. 

I took a slightly different approach to the implementation than @michaelrog suggested though in that I refactored to use a template by default. This was we avoid branching and just consume our own config settings in the default template so as to keep compatibility with the old config interface.

The new `getRenderedTemplate()` service method handles either a (CP) template path or a template string, so you can just use a `HEREDOC`in the config file if you're feeling dirty, or pass a path to override the default.

I've also taken the opportunity to clean generally things up a bit, and there are enough changes that would appreciate any comments / quick testing you could offer.

@michaelrog no need to worry about the login screen leaking information - we already check for a logged in user rather than just a CP request.

**Comment by khalwat** [comment]
So overall, I love it. My feedback thus far re: the design is... I like that you're using a template that can be customized to be whatever.

My only issue with it is that:

1) I don't really want to make a plugin just to have this template be available

2) I don't really want to expose the template in the `templates` folder, because it in theory is then accessible publicly.

3) I don't really want to pass in the entire template + CSS as a string because that's a little gross.

None of this is an issue if the actual `label` that is passed in is the thing that is rendered as a template, and then we get the nice wrapper you've provided around all of it. The formatting controls that already exist are find, I really just want a different color and name per env, I can't imagine anyone going too overboard with a customized template for this.

So for now, what I'm planning to do is just go with option 2) and wrap it in `{% requireLogin %}` so that they at least need to be logged in to see it the template, it they do somehow stumble across the public URL to the template.

**Comment by khalwat** [comment]
Ah, it just occurred to me that I could just prefix the public template with an underscore, to make it not renderable directly. Trying...

(might be something for the docs as a recommended best practice; you might even name the default template `_default.html` or such)

**Comment by khalwat** [comment]
Alright, turns out I misunderstood your comments, and actually the template _has_ to be either a plugin template, or a string.

So for now, I'm just modifying the Environment Label `default.html` template itself. Which I realize is a pretty terrible way to do it but...

**Comment by khalwat** [comment]
Ack... with the 3.0 dev branch plugin installed, frontend requests are all 500 errors (the AdminCP works fine, however):

![google chromescreensnapz321](https://cloud.githubusercontent.com/assets/7570798/22315320/07606b50-e334-11e6-93a9-b6af9a1611cd.png)

...I'll try to see if I can't figure out what's gone wrong here.

**Comment by khalwat** [comment]
Yeah so the issue is your calling:

    'renderedTemplate' => craft()->environmentLabel->getRenderedTemplate(),

...inside of your TwigExtension... because that causes it to render a template, which re-instantiates your TwigExtension, which calls it again... well, you get the idea. Infinite loop until PHP goes tits up.

I'd suggest something like this, to be safe...

```
class EnvironmentLabelTwigExtension extends \Twig_Extension
{

    protected $environmentLabelInitializing = false;

    /**
     * Returns an array to be merged into Twig's global variables.
     *
     * @return array The globals defined by this extension
     */
    public function getGlobals()
    {

        $globals = array();
        if (!$this->environmentLabelInitializing && craft()->request->isCpRequest() && craft()->userSession->isLoggedIn()) {
            $this->environmentLabelInitializing = true;
            $globals['environmentLabel'] = array(
                'showLabel' => craft()->environmentLabel->getShowLabel(),
                'label' => craft()->environmentLabel->getLabel(),
                'prefix' => craft()->environmentLabel->getPrefix(),
                'suffix' => craft()->environmentLabel->getSuffix(),
                'labelColor' => craft()->environmentLabel->getLabelColor(),
                'textColor' => craft()->environmentLabel->getTextColor(),
                'renderedTemplate' => craft()->environmentLabel->getRenderedTemplate(),
            );
        }
        $this->environmentLabelInitializing = false;
        return $globals;
    }
```
